proposals/1984_ca-bundle-from-secret-ref/README.md

```diff
@@ -11,124 +11,261 @@ Once approved and implemented, they become historical documents.
+      -----BEGIN CERTIFICATE-----
+      MIIOfDC...
+      -----END CERTIFICATE-----
+  target:
+    secret:
+      key: "root-certs.pem"
 
-### Supervisor
+# results in a secret with name some-pinniped-test
+apiVersion: v1
+data:
+  root-certs.pem: something-base64-encoded
+kind: Secret
+metadata:
+  labels:
+    trust.cert-manager.io/bundle: some-pinniped-test
+  name: some-pinniped-test
+  namespace: concierge
+  ownerReferences:
+  - apiVersion: trust.cert-manager.io/v1alpha1
+    blockOwnerDeletion: true
+    controller: true
+    kind: Bundle
+    name: some-pinniped-test
+    uid: 5bf1b44f-9642-4335-8f76-d16f73a0329f
+  resourceVersion: "79068"
+  uid: f71829b9-ee06-466c-b7ec-43a07b6852bc
+type: Opaque
+```
 
-* `ActiveDirectoryIdentityProvider.spec.tls.certificateAuthorityData`
-  * Code: `internal/controller/supervisorconfig/activedirectoryupstreamwatcher/active_directory_upstream_watcher.go`
-  * Status Condition: `TLSConfigurationValid`
-* `GitHubIdentityProvider.spec.githubAPI.tls.certificateAuthorityData`
-  * Code: `internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher.go`
-  * Status Condition: `TLSConfigurationValid`
-* `LDAPIdentityProvider.spec.tls.certificateAuthorityData`
-  * Code: `internal/controller/supervisorconfig/ldapupstreamwatcher/ldap_upstream_watcher.go`
-  * Status Condition: `TLSConfigurationValid`
-* `OIDCIdentityProvider.spec.tls.certificateAuthorityData`
-  * Code: `internal/controller/supervisorconfig/oidcupstreamwatcher/oidc_upstream_watcher.go`
-  * Status Condition: currently `OIDCDiscoverySucceeded` appears to be the applicable status condition, but we should
-  add a new status condition `TLSConfigurationValid` to resemble the other identity provider CRs.
+Below is an example of trust-manager's bundle resource and the corresponding target of type `configMap`.
 
-### Concierge
+```bash
+apiVersion: trust.cert-manager.io/v1alpha1
+kind: Bundle
+metadata:
+  name: some-pinniped-test-cm
+spec:
+  sources:
+  - inLine: |
+      -----BEGIN CERTIFICATE-----
+      MIIOfDCC...
+      -----END CERTIFICATE-----
+  target:
+    configMap:
+      key: "foo.bar.baz"
+
+# results in a configMap with name some-pinniped-test-cm
+
+apiVersion: v1
+data:
+  foo.bar.whatever: |
+    -----BEGIN CERTIFICATE-----
+    MIIOfDCC...
+    -----END CERTIFICATE-----
+kind: ConfigMap
+metadata:
+  labels:
+    trust.cert-manager.io/bundle: some-pinniped-test-cm
+  name: some-pinniped-test-cm
+  namespace: concierge
+  ownerReferences:
+  - apiVersion: trust.cert-manager.io/v1alpha1
+    blockOwnerDeletion: true
+    controller: true
+    kind: Bundle
+    name: some-pinniped-test-cm
+    uid: ca2fe559-fb0b-4bb0-ae75-1b9b43df1d4c
+  resourceVersion: "82195"
+  uid: 08c96a5a-f59b-4e76-b0e2-7132abe16411
+```
+
+#### Vault
+
+Based on the usage reported in [issues/1886](https://github.com/vmware-tanzu/pinniped/issues/1886) the CA trust bundle is externally sourced into the kubernetes cluster as a secret.
```

Comment:
I suspect that user is using Vault's Kubernetes Secrets Operator. It can create Secrets of various types, but by default will create a Secret of type Opaque, according to the docs. See https://developer.hashicorp.com/vault/docs/platform/k8s/vso/api-reference#destination. That seems compatible with this proposed solution.

We could update this section to discuss these points instead of pointing to #1886 which is pretty vague about the format of the Secret created by vault.